battle-engine.js

```diff
@@ -4676,7 +4676,7 @@ exports.BattleSide = BattleSide;
 exports.Battle = Battle;
 
 let battleProtoCache = new Map();
-exports.construct = function (format, rated, send) {
+exports.construct = function (format, rated, supplementaryRuleset, send) {
```

Comment:
A `send` function is so passé though...

I think maybe the format should be constructed outside of this function, so this function's signature stays unchanged.